daffodil-runtime2/src/main/resources/c/libcli/xml_reader.c

```diff
@@ -97,6 +98,72 @@ strtounum(const char *numptr, uintmax_t maxval, const char **errstrp)
     return value;
 }
 
+// Convert an XML element's text to a float (call strtof with our own
+// error checking)
+
+static float
+strtofnum(const char *numptr, const char **errstrp)
```

Comment:
I see; that makes sense.  Instead of checking whether a const char *error_msg was returned, we can pass an error_info struct and check whether it has some error messages afterwards.  I don't think Daffodil has internationalized messages right now; has anyone asked for that support soon?

Reply:
Internationalized message support will eventually come up. It's very painful unless we put the support in early. This is also an issue of performance for backtracking (eventually someday)